app/services/publication-service.js

```diff
@@ -64,4 +64,22 @@ export default class PublicationService extends Service {
     const publicationNumberTakenList = publicationWithId.filter((publicationFlow) => publicationFlow.id !== publicationFlowId);
     return publicationNumberTakenList.toArray().length !== 0;
   }
+
+  async getNewPublicationNextNumber(isViaMinisterCouncil) {
+    if (isViaMinisterCouncil) {
+      // nextnumber generation implementation for via minister council
+    } else {
+      return (await this.store.query('publication-flow', {
+        filter: {
+          ':has:case': 'yes',
+          case: {
+            ':has-no:subcases': 'yes',
+          },
+          status: {
+            id: CONFIG.publicationStatusToPublish.id,
+          },
+        },
+      })).length + 1;
+    }
+  }
```

Comment:
Dit lijkt me niet de gewenste implementatie.. wat ik hier vermoed is dat je enkel de publicaties ophaalt die dat status hebben om te publishen.. je wilt hier echter alle publicaties ophalen en vervolgens aflopend sorteren.
Daarna neem je het eerste element en doe je dat +1, dan heb je het volgende getal dat je nodig hebt.

De if/else is in dit geval dus ook overbodig.. je dient de code zo te schrijven dat het gewoon het eerste nummer teruggeeft, ongeacht of het via ministerraad gaat of niet.

Je dient dit ook te implementeren voor `via ministerraad` hier ga je in de code zien dat er een placeholder staat met `0` als ID.. hier dien je dan het resultaat van deze query in te steken.